src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/ApiBehaviorConventions/InferModelPrefixConvention.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationModels
+{
+    /// <summary>
+    /// An <see cref="IControllerModelConvention"/> that infers <see cref="BindingInfo.BinderModelName"/>
+    /// for bound properties and parameters.
+    /// </summary>
+    public class InferModelPrefixConvention : IControllerModelConvention
+    {
+        private readonly IModelMetadataProvider _modelMetadataProvider;
+
+        public InferModelPrefixConvention(IModelMetadataProvider modelMetadataProvider)
```

Comment:
These are still problematic because they need a `IModelMetadataProvider`, but not much we could do about it.